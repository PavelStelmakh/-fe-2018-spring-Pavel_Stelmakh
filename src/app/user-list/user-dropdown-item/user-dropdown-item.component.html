<div>
  <div>
    <span>{{'user-details.output-form.name' | translate}}:</span>
    <span>{{user?.name}}</span>
  </div>
  <div>
      <span>{{'user-details.output-form.age' | translate}}:</span>
      <span>{{user?.age}}</span>
  </div>
  <div>
      <span>{{'user-details.output-form.birthday' | translate}}:</span>
      <span>{{user?.dateOfBirth}}</span>
  </div>
  <div>
      <span>{{'user-details.output-form.dateOfLogin' | translate}}:</span>
      <span>{{user?.dateOfFirstLogin}}</span>
  </div>
  <div>
    <span>{{'user-details.output-form.information' | translate}}:</span>
    <span>{{user?.information}}</span>
</div>
  <div>
      <span>{{'user-details.output-form.dateOfNotification' | translate}}:</span>
      <span>{{user?.dateOfNextNotif}}</span>
  </div>
</div>
