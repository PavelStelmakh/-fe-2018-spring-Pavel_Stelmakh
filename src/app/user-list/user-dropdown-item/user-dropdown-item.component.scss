:host {
    display: block;
    width: 500px;
    height: 150px;
    display: flex;
    justify-content: space-around;
    align-items: center;
    border-bottom: 1px solid #ccc;
    text-align: start;

    & > div > div {
        width: 400px;
        display: flex;
        justify-content: space-between;
    }

}