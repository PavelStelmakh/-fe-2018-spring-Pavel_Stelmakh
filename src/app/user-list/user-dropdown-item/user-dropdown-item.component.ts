import { Component, OnInit, Input } from '@angular/core';
import { User } from '../../../../models/User';
import * as moment  from 'moment';

@Component({
  selector: 'app-user-dropdown-item',
  templateUrl: './user-dropdown-item.component.html',
  styleUrls: ['./user-dropdown-item.component.scss']
})
export class UserDropdownItemComponent implements OnInit {
  @Input() user: User;

  constructor() { }

  ngOnInit() {
    const dateOfNextNotif = moment(this.user.dateOfNextNotif).utcOffset('+04:00').format('DD-MMM-YY');
    this.user.dateOfBirth = moment(this.user.dateOfBirth).utcOffset('+04:00').format('YYYY/MM/DD');
    this.user.dateOfFirstLogin = moment(this.user.dateOfFirstLogin).utcOffset('+04:00').format('DD MMMM YYYY');
    this.user.dateOfNextNotif = dateOfNextNotif !== 'Invalid date' ? dateOfNextNotif: this.user.dateOfNextNotif;
  }

}
