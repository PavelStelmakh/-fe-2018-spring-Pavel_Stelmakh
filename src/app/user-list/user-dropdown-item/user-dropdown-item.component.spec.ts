import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Observable, of } from 'rxjs';
import { TranslateModule, TranslateLoader, TranslateService } from '@ngx-translate/core';
import { UserDropdownItemComponent } from './user-dropdown-item.component';

const translations = {
  "user-details": {
    "output-form": {
      "name": "Name",
      "password": "Password",
      "age": "Age",
      "birthday": "Birthday",
      "dateOfLogin": "Date of login",
      "information": "Information",
      "dateOfNotification": "Date of notification"
    }
  }  
};

class FakeLoader implements TranslateLoader {
  getTranslation(lang: string): Observable<{}> {
    return of(translations);
  }
}

describe('UserDropdownItemComponent', () => {
  let component: UserDropdownItemComponent;
  let fixture: ComponentFixture<UserDropdownItemComponent>;
  let translate: TranslateService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        TranslateModule.forRoot({
          loader: {provide: TranslateLoader, useClass: FakeLoader}
        })
      ],
      declarations: [ UserDropdownItemComponent ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserDropdownItemComponent);
    component = fixture.componentInstance;
    translate = TestBed.get(TranslateService);
  });

  it('renders markup to snapshot', () => {
    component.user = {
      id: 2,
      name: 'Pasha',
      age: 19,
      password: '1111',
      dateOfBirth: '2011-10-05T00:00:00.000Z',
      dateOfFirstLogin: '2011-10-05T14:48:00.000Z',
      dateOfNextNotif: '2011-10-05T14:48:00.000Z',
      information: 'meaw',
      role: 'user'
    };
    translate.use('en');
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  // it('should create', () => {
  //   expect(component).toBeDefined();
  // });
});
