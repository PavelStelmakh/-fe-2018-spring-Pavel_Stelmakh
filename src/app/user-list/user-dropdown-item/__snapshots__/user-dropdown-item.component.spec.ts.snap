// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDropdownItemComponent renders markup to snapshot 1`] = `
<app-user-dropdown-item
  user={[Function Object]}
>
  <div>
    <div>
      <span>
        Name:
      </span>
      <span>
        Pasha
      </span>
    </div>
    <div>
      <span>
        Age:
      </span>
      <span>
        19
      </span>
    </div>
    <div>
      <span>
        Birthday:
      </span>
      <span>
        2011/10/05
      </span>
    </div>
    <div>
      <span>
        Date of login:
      </span>
      <span>
        05 October 2011
      </span>
    </div>
    <div>
      <span>
        Information:
      </span>
      <span>
        meaw
      </span>
    </div>
    <div>
      <span>
        Date of notification:
      </span>
      <span>
        05-Oct-11
      </span>
    </div>
  </div>
</app-user-dropdown-item>
`;
